starknet_py/tests/e2e/docs/code_examples/test_gateway_client.py

```diff
@@ -0,0 +1,207 @@
+# pylint: disable=unused-variable, redefined-builtin
+import pytest
+from starkware.starknet.public.abi import (
+    get_selector_from_name,
+    get_storage_var_address,
+)
+
+from starknet_py.net.client_models import Call
+from starknet_py.net.gateway_client import GatewayClient
+from starknet_py.net.networks import TESTNET
+from starknet_py.tests.e2e.fixtures.constants import MAX_FEE
+
+
+def test_init():
+    # docs-start: init
+    gateway_client = GatewayClient(net=TESTNET)
+    # or (for custom urls)
+    gateway_client = GatewayClient(
+        net={"feeder_gateway_url": "some_feeder_url", "gateway_url": "some_gateway_url"}
+    )
+    # docs-end: init
+
+
+@pytest.mark.asyncio
+async def test_get_block(gateway_client):
+    # docs-start: get_block
+    block = await gateway_client.get_block(block_hash="latest")
+    # or
+    block = await gateway_client.get_block(block_number=0)
+    # docs-end: get_block
+
+
+@pytest.mark.asyncio
+async def test_get_block_traces(gateway_client):
+    # docs-start: get_block_traces
+    block_traces = await gateway_client.get_block_traces(block_hash="latest")
```

Comment:
Let's change it to `block_number` as the gateway spec doesn't allow for the block_hash to be `latest`, and add another example for `block_hash` keyword